### PR TITLE
#25 Investors Design

### PR DESCRIPTION
**Pull request by Nada257**

### CODE REVIEW

File: retail-managment-system/rms-ui/src/app/domain/investors/models/investor.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export interface Investors {
+  name: any;
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/domain/investors/models/investor.ts:2]: why add name ?

File: retail-managment-system/rms-ui/src/app/domain/investors/models/investor.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  id: number;
+  fullName: string;
+  nickName: string;
+  nationalId: string;
+  primaryPhoneNo: number;
+  secondaryPhoneNo: number;
+  address: string;
+  investorType: string;
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/domain/investors/models/investor.ts:10]: investorType is enum in swagger not string

File: retail-managment-system/rms-ui/src/app/domain/investors/models/investor.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  balance: number;
+  startDate: number;
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/domain/investors/models/investor.ts:12]: is startDate number in swagger?

File: retail-managment-system/rms-ui/src/app/pages/investor-page/investor-page.component.html [hunk 4]
```diff
@@ -0,0 +1,81 @@
+<div class="container-fluid">
+  <br /><br />
+  <div class="row">
+    <div class="col-lg-10 offset-lg-1">
+      <div class="card-body">
+        <div class="row">
+          <div class="col-12">
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/investor-page/investor-page.component.html:7]: are you need all this div ?

File: retail-managment-system/rms-ui/src/app/pages/investor-page/investor-page.component.html [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            </div>
+          </div>
+        </div>
+        <br />
+        <div class="row">
+          <div class="col-lg-15 offset-lg-0">
+            <div class="card">
+              <div class="card-body">
+                <div class="row">
+                  <table class="table table-centered mb-0">
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/investor-page/investor-page.component.html:28]: why didn't used material table ?

File: retail-managment-system/rms-ui/src/app/pages/investor-page/pop-up/pop-up.component.ts [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+      id: [''],
+      fullName: ['', [Validators.required]],
+      nickName: ['', [Validators.required]],
+      nationalId: ['', [Validators.required]],
+      primaryPhoneNo: ['', [Validators.required]],
+      secondaryPhoneNo: ['', [Validators.required]],
+      address: ['', [Validators.required]],
+      investorType: ['', [Validators.required]],
+      balance: ['', [Validators.required]],
+      startDate: ['', [Validators.required]],
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/investor-page/pop-up/pop-up.component.ts:36]: why you make all controls required depending on what?
and 
phone must be 11  number 
email like emails 
nationalId must be 14 number

**Review by ZinabHashem — CHANGES_REQUESTED** [review]

File: retail-managment-system/rms-ui/src/app/pages/investor-page/pop-up/pop-up.component.ts (hunk 6 above)
Comment (Nada257) [reply retail-managment-system/rms-ui/src/app/pages/investor-page/pop-up/pop-up.component.ts:36]: what should be required, and what should not?

File: retail-managment-system/rms-ui/src/app/pages/investor-page/investor-page.component.html (hunk 4 above)
Comment (Nada257) [reply retail-managment-system/rms-ui/src/app/pages/investor-page/investor-page.component.html:7]: Removed card body , but row used to organize the code and separate the first row to the second

File: retail-managment-system/rms-ui/src/app/pages/investor-page/pop-up/pop-up.component.ts (hunk 6 above)
Comment (ZinabHashem) [reply retail-managment-system/rms-ui/src/app/pages/investor-page/pop-up/pop-up.component.ts:36]: in general, from swagger object 
but now it not handled yet
so you can ask for that before make your that


Temporarily, you can make investorType, fullName, nationalId and primaryPhoneNo required

File: retail-managment-system/rms-ui/src/app/pages/investor-page/pop-up/pop-up.component.html [hunk 7]
```diff
@@ -64,7 +64,9 @@ <h3 class="panel-title">Manage Investors</h3>
                 type="number"
                 matInput
                 placeholder="Enter nationalId"
-                fxFlex="14"
+                fxFlex="40"
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/investor-page/pop-up/pop-up.component.html:67]: is this work ?

File: retail-managment-system/rms-ui/src/app/pages/investor-page/pop-up/pop-up.component.html [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+                Investor address is <strong>required</strong>
+              </mat-error>
+            </mat-form-field>
+            <mat-form-field appearance="outline" class="m-2">
+              <mat-label> investorType</mat-label>
+              <input
+                formControlName="investorType"
+                type="string"
+                matInput
+                placeholder="Enter investorType Manager or CONTRIBUTOR"
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/investor-page/pop-up/pop-up.component.html:188]: how you check that client enter right word?

**Review by ZinabHashem — CHANGES_REQUESTED** [review]
are you test that number didn't accept more than or less than 11?
same in national id?

please test all your code like (cycle and validations and ......)  before push

File: retail-managment-system/rms-ui/src/app/core/layout/footer/footer.component.html [hunk 9]
```diff
@@ -1,5 +1,5 @@
 <div class="copyright-footer d-flex align-items-center justify-content-center">
   <div class="row">
-    <span class="text-white fs-6">@2020. All rights reserved.</span>
+    <span class="text-white fs-50">@2020. All rights reserved.</span>
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/core/layout/footer/footer.component.html:3]: what is this change?

File: retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.ts [hunk 10]
```diff
@@ -1,40 +1,73 @@
...
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { CategoryRepository } from '../../domain/category/category.repository';
 import { Category } from 'src/app/domain/category/models/category';
-import { MatDialog } from '@angular/material/dialog';
-import { PopUpComponent } from './pop-up/pop-up.component';
 @Component({
   selector: 'app-category-page',
   templateUrl: './category-page.component.html',
+  styles: [
+    '.category { min-height: auto; } table { min-width: 1100px; min-height: auto; }mat-icon{ font-size: 29px;} .btn {background-color: #002d40; color: white;}',
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.ts:9]: return file css if you used a lot of style

File: retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.ts [hunk 11]
```diff
@@ -1,40 +1,73 @@
...
+  ],
 })
 export class CategoryPageComponent implements OnInit {
+  categoryform!: FormGroup;
   allCategory: Category[] = [];
   isAppear!: boolean;
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.ts:15]: are you used this property?

File: retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.ts [hunk 12]
```diff
@@ -1,40 +1,73 @@
...
-  displayedColumns: string[] = ['id', 'name', 'status'];
-  dataimage: any;
+  displayedColumns: string[] = ['id', 'name', 'status', 'update', 'delete'];
   submit: boolean = false;
   submitted: boolean = false;
+  data!: Category;
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.ts:19]: are you used this property?
line 18 and 19?

File: retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.html [hunk 13]
```diff
@@ -1,67 +1,123 @@
...
-                  </td>
-                </tr>
-              </tbody>
-            </table>
+                <button
+                  class="btn me-1 mb-1"
+                  (click)="resetTheForm()"
+                  mat-button
+                >
+                  Reset
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.html:114]: this button reset on update 
where clear if add?

File: retail-managment-system/rms-ui/src/app/pages/category-page/pop-up/pop-up.component.html [hunk 14]
```diff
@@ -6,14 +6,14 @@ <h3 class="panel-title">Manage Category</h3>
       </div>
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/category-page/pop-up/pop-up.component.html:6]: why didn't deleted popup?

File: retail-managment-system/rms-ui/src/app/pages/investor-page/investor-page.component.html [hunk 15]
```diff
@@ -0,0 +1,359 @@
...
+            </fieldset>
+          </form>
+          <mat-dialog-actions align="end">
+            <div class="row">
+              <div class="col-12">
+                <div class="page-footer-box">
+                  <button
+                    class="btn btn-success me-2"
+                    (click)="onSubmit()"
+                    (click)="onButtonClick()"
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/investor-page/investor-page.component.html:338]: what is this function?

File: retail-managment-system/rms-ui/src/app/pages/product/product.component.html [hunk 16]
```diff
@@ -34,7 +34,7 @@ <h3 class="panel-title fw-bold">Our Products</h3>
 
         <ng-container matColumnDef="price" class="price">
           <th mat-header-cell *matHeaderCellDef class="fw-bold fs-6">Price</th>
-          <td mat-cell *matCellDef="let element">{{ element.cashPrice }}</td>
+          <td mat-cell *matCellDef="let element">{{ element.shPrice }}</td>
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/product/product.component.html:37]: why changed this?

**Review by ZinabHashem — CHANGES_REQUESTED** [review]

File: retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.ts (hunk 11 above)
Comment (Nada257) [reply retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.ts:15]: yes at Fetch, add, and update functions

File: retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.ts (hunk 12 above)
Comment (Nada257) [reply retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.ts:19]: Submitted used in onsubmit function, but data!:Category deleted

**Pull request merged**